### ISSUE TITLE
CIMInstanceName.from_wbemuri fails one of the allowed formats

### ISSUE THREAD

**Issue by KSchopmeyer** [pywbem/pywbem#1359]
It appears that the wbemuri parser does not allow the format

classname.keybindings and requires the leading ":" before the classname.

I am running tests in pywbemcli after moving to using this as the parser and the following fails

'PyWBEM_AllTypes.InstanceID="test_instance"'

E       Error: Invalid wbem uri input PyWBEM_AllTypes.InstanceID="test_instance". Error Invalid format for an instance path in WBEM URI: 'PyWBEM_AllTypes.InstanceID="test_instance"'

Simply add the : before the classname and the from_wbemuri passes.

**Comment by andy-maier** [comment]
That's weird. We do have a test case `CIM_Foo.k1="v1"`.

**Comment by andy-maier** [comment]
@KSchopmeyer I added the WBEM URI `PyWBEM_AllTypes.InstanceID="test_instance"` to the testcases for `CIMInstanceName.from_wbem_uri()` and it succeeded. I was using the current master version of pywbem (0.12.1.dev78).

Could you please double check whether you are using `CIMInstanceName.from_wbem_uri()`, and which version of pywbem you are using?

**Comment by andy-maier** [comment]
I believe the reason for this is that you did not yet pick up the fix for tolerating the WBEM URI without leading colon. That fix has meanwhile been merged to pywbem master.

**Comment by andy-maier** [comment]
The code in the pywbemtools project that currently circumvents this issue is in [parse_wbemuri_str()](https://github.com/pywbem/pywbemtools/blob/ks/add-modify-inst/pywbemcli/_common.py#L271).

**Comment by andy-maier** [comment]
@KSchopmeyer My only recommendation at this point is that you add some debug prints in your circumvention code in parse_wbemuri_str().

**Comment by KSchopmeyer** [comment]
**DISCUSSION:** I believe this one should go into a 0.12.6 update but lets wait a few days to see what else pops up.

I removed myself as assignee since 1. We think the bug is fixed in 0.13.0 and therefore, the only issue is backing it into 0.12.6.

**Comment by andy-maier** [comment]
Karl: SInce I was not able to reproduce the error with the latest master, please reproduce this, report exactly what happened and what the string was, and particularly report the pywbem dev version you are using.

**Comment by andy-maier** [comment]
Given your last comment:

* "COMMENT:I also manually tested using wbemcli and agree that the bug is fixed in the master now."

Can we close this issue?

**Comment by andy-maier** [comment]
The manual testcase using just the pywbem API is:

```
$ python -c "import pywbem; print(repr(pywbem.CIMInstanceName.from_wbem_uri('CIM_Foo.ID=1')))"
CIMInstanceName(classname=u'CIM_Foo', keybindings=NocaseDict([('ID', 1)]), namespace=None, host=None)
```

**Comment by andy-maier** [comment]
COnclusion was that Karl runs the test again.

**Comment by andy-maier** [comment]
New conclusion after the investigation shown two comments further up is that this was fixed in 0.13.0. Pywbemtools should continue with the workaround and remove the workaround once pywbem 0.13.0 is released.

Suggest to close this issue.

**Comment by andy-maier** [comment]
We agreed in the call today to close this issue.
